packages/flutter/lib/src/widgets/form.dart

```diff
@@ -291,18 +291,44 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  ///
+  /// See also:
+  ///  * [validateGranularly], which returns a [Map] describing the validation
+  /// status for each [FormField]
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
     return _validate();
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField]s that is a descendant of this [Form], and
+  /// returns a [List] of [FormFieldState] of the invalid field(s) only, if any.
```

Comment:
Does the order matter? Should we return an unordered collection instead?

Reply:
The order could matter. It doesn't have to be predictable but has to be consistent. i.e. `validateGranularly().first` should return the same field as long as it's still invalid.

I replaced List with Set, but I don't think we can replace it with something not ordered like `HashSet`. Let me know if you have other collections in mind.

The rest of the comments are addressed too.